azure-mgmt-network/src/main/java/com/microsoft/azure/management/network/implementation/PointToSiteConfigurationImpl.java

```diff
@@ -0,0 +1,107 @@
+ */
+@LangDefinition
+class PointToSiteConfigurationImpl
+        extends ChildResourceImpl<VpnClientConfiguration, VirtualNetworkGatewayImpl, VirtualNetworkGateway>
+        implements
+        PointToSiteConfiguration,
+        PointToSiteConfiguration.Definition<VirtualNetworkGateway.Update>,
+        PointToSiteConfiguration.Update {
+
+    PointToSiteConfigurationImpl(VpnClientConfiguration inner, VirtualNetworkGatewayImpl parent) {
+        super(inner, parent);
+    }
+
+    @Override
+    public String name() {
+        return null;
+    }
+
+    @Override
+    public VirtualNetworkGatewayImpl attach() {
+        this.parent().attachPointToSiteConfiguration(this);
+        return parent();
+    }
+
+    @Override
+    public PointToSiteConfigurationImpl withAddressPool(String addressPool) {
+        List<String> addressPrefixes = new ArrayList<>();
+        addressPrefixes.add(addressPool);
+        inner().withVpnClientAddressPool(new AddressSpace().withAddressPrefixes(addressPrefixes));
+        return this;
+    }
+
+    @Override
+    public PointToSiteConfigurationImpl withRootCertificate(String name, String certificateData) {
+        if (inner().vpnClientRootCertificates() == null) {
+            inner().withVpnClientRootCertificates(new ArrayList<VpnClientRootCertificateInner>());
+        }
+        inner().vpnClientRootCertificates().add(new VpnClientRootCertificateInner().withName(name).withPublicCertData(certificateData));
+        return this;
+    }
+
+    @Override
+    public Update withoutRootCertificate(String name) {
+        if (inner().vpnClientRootCertificates() != null) {
+            for (VpnClientRootCertificateInner certificateInner : inner().vpnClientRootCertificates()) {
+                if (name.equals(certificateInner.name())) {
+                    inner().vpnClientRootCertificates().remove(certificateInner);
+                    break;
+                }
+            }
+        }
+        return this;
+    }
+
+    @Override
+    public PointToSiteConfigurationImpl withAzureCertificate() {
+        return this;
+    }
+
+    @Override
+    public PointToSiteConfigurationImpl withRadiusAuthentication(String serverIPAddress, String serverSecret) {
+        inner().withRadiusServerAddress(serverIPAddress).withRadiusServerSecret(serverSecret);
+        return this;
+    }
+
+    @Override
+    public PointToSiteConfigurationImpl withRevokedCertificate(String name, String thumbprint) {
+        if (inner().vpnClientRevokedCertificates() == null) {
+            inner().withVpnClientRevokedCertificates(new ArrayList<VpnClientRevokedCertificateInner>());
+        }
+        inner().vpnClientRevokedCertificates().add(new VpnClientRevokedCertificateInner().withName(name).withThumbprint(thumbprint));
+        return this;
+    }
+
+    @Override
+    public PointToSiteConfigurationImpl withSstpOnly() {
+        inner().withVpnClientProtocols(Collections.singletonList(VpnClientProtocol.SSTP));
```

Comment:
I've quickly looked at the docs and I only saw that these properties are set/relevant for the RADIUS connections. Can you test if the create goes through if set for an Azure connection certification?

Reply:
Portal allows to specify in both cases.